.github/workflows/e2e-ci.yaml

```diff
@@ -182,6 +183,37 @@ jobs:
       env:
         CONTAINER_REGISTRY_ADDRESS: ${{ env.CONTAINER_REGISTRY_ADDRESS }}
 
+  # Generate root and server certificates and copy required files to RPi
+  certificates_job :
+    timeout-minutes: 5
+    name: Generate and copy certificates
+    environment:
+      name: CI_AZURE_ENVIRONMENT
+    if: needs.env_var.outputs.RunE2ETestsOnly != 'true' && needs.env_var.outputs.StopFullCi != 'true'
+    needs:
+      - env_var
+    runs-on: [ self-hosted, x64 ]
+
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout current branch
+
+    - id: generate_step
+      shell: bash
+      name: Generate server and client certificates
+      env:
+        SERVER_PFX_PASSWORD: ${{ secrets.SERVER_PFX_PASSWORD }}
+        SSH_PRIVATE_KEY_PATH: ${{ secrets.SSH_PRIVATE_KEY_PATH }}
```

Comment:
Should we add a brief description what this secret is? (I assume we need the cert to be present on the Github runner machine to connect to the Pi that we use for E2E tests? (same goes for the other secrets that were introduced, such as `RPI_USERNAME`)

Reply:
We should probably (as you suggested @bastbu) start documenting the CI and the different variable. This is really starting to become quite complex

Reply:
Created #1015 to follow up on that